src/mango/src/mango_cursor_text.erl

```diff
@@ -109,7 +110,7 @@ execute(Cursor, UserFun, UserAcc) ->
         user_fun = UserFun,
         user_acc = UserAcc,
         fields = Cursor#cursor.fields,
-        execution_stats = mango_execution_stats:log_start(Stats),
+        execution_stats = mango_execution_stats:log_start(Stats, DbName),
```

Comment:
I think the `create` family of functions might be a better place to initialize the `#execution_stats{}` record with the database name immediately.  So that you would not need add it in `execute/3` and `mango_execution_stats:log_start/1` would not need an extra parameter for that.  For example, in case of `mango_cursor_text`, something along these lines:

```diff
diff --git a/src/mango/src/mango_cursor_text.erl b/src/mango/src/mango_cursor_text.erl
index ee1d962f7..5cc06344f 100644
--- a/src/mango/src/mango_cursor_text.erl
+++ b/src/mango/src/mango_cursor_text.erl
@@ -50,7 +50,8 @@ create(Db, {Indexes, Trace}, Selector, Opts0) ->
                 ?MANGO_ERROR(multiple_text_indexes)
         end,

-    Opts = unpack_bookmark(couch_db:name(Db), Opts0),
+    DbName = couch_db:name(Db),
+    Opts = unpack_bookmark(DbName, Opts0),

     DreyfusLimit = get_dreyfus_limit(),
     Limit = erlang:min(DreyfusLimit, couch_util:get_value(limit, Opts, mango_opts:default_limit())),
@@ -66,7 +67,8 @@ create(Db, {Indexes, Trace}, Selector, Opts0) ->
         opts = Opts,
         limit = Limit,
         skip = Skip,
-        fields = Fields
+        fields = Fields,
+        execution_stats = #execution_stats{dbname = DbName}
     }}.

 explain(Cursor) ->
```

Reply:
Yeah good call on moving this into the create logic, that seems better. The initialization of the `#execution_stats{}` record is a bit convoluted though in that it happens in `mango_cursor.hrl` here https://github.com/apache/couchdb/blob/4643fec27b2cd75a504c7c6356f60c0127acb4a5/src/mango/src/mango_cursor.hrl#L28 and your patch actually creates a brand new version of the `#execution_stats{}` record, dumping the original one.

I've taken your suggestion of moving this logic into the create calls, and I've also reworked the `#execution_stats{}` initialization logic to be postponed until the create function as well, along with utilize a dedicated initializer function in 517fce8b6.

Reply:
And I've also unwound the inclusion of dbname in `log_start` as part of that commit too, so we can see the partial delta, look at the PR view instead of the commit view to exclude the intermediary stage code.